src/Main.ps1

```diff
@@ -922,6 +922,25 @@ function Invoke-Pester {
             $pluginConfiguration = @{}
             $pluginData = @{}
             $plugins = @()
+
+            # Verify this before WriteScreenPlugin because of Write-PesterStart and Write-PesterDebugMessage
+            if ($PesterPreference.Output.RenderMode.Value -notin 'Auto', 'Ansi', 'Legacy', 'Plaintext') {
+                throw "Unsupported Output.RenderMode option '$($PesterPreference.Output.RenderMode.Value)'"
+            }
+
+            if ($PesterPreference.Output.RenderMode.Value -eq 'Auto') {
+                if ($null -ne $env:NO_COLOR) {
+                    # https://no-color.org/)
+                    $PesterPreference.Output.RenderMode = 'Plaintext'
+                }
+                elseif (($supportsVT = $host.UI.psobject.Properties['SupportsVirtualTerminal']) -and $supportsVT.Value) {
+                    $PesterPreference.Output.RenderMode = 'Ansi'
+                }
+                else {
+                    $PesterPreference.Output.RenderMode = 'Legacy'
```

Comment:
Not a big fan of "Legacy" options over concrete option name, do we use similar naming in other places as well?  I can't remember.

Reply:
Only parameter set name of the top of my head.

Suggestions? `HostOnly` or `Console` felt like still could be ANSI and possible complaint. In case we add other styles in the future like bold, underline etc.

`ConsoleColor` was a bit technical, but maybe better? `Compatibility` maybe?

Reply:
I think ConsoleColor is the one to use, it is a bit technical, but we can explain in docs somewhere in 1 line. The meaning of "Legacy" will just deteriorate, and then soon there will be Legacy2, or Legacy will be used for 10 different purposes.